File: web/lib/internal-dossier.ts
import { apiJson } from "@/lib/api";

export type InternalDossierRateCard = {
  rate_card_id: string | null;
  name: string | null;
  version_number: number | null;
  effective_date: string | null;
  settings: Record<string, unknown>;
};

export type InternalDossierProposal = {
  id: string;
  locale: string;
  status: string;
  include_poc: boolean;
  assessment: Record<string, unknown> | null;
  proposal_body: Record<string, unknown> | null;
  poc: Record<string, unknown> | null;
};

export type InternalDossier = {
  estimate_id: string;
  project_name: string;
  client_name: string;
  status: string;
  locale: string;
  has_calculation: boolean;
  rate_card_stale: boolean;
  warnings: string[];
  report: Record<string, unknown>;
  rate_card: InternalDossierRateCard | null;
  proposals: InternalDossierProposal[];
};

export async function fetchInternalDossier(
  estimateId: string,
): Promise<InternalDossier> {
  return apiJson<InternalDossier>(`/estimates/${estimateId}/internal-dossier`);
}

File: api/app/schemas/internal_dossier.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class InternalDossierRateCard(BaseModel):
    rate_card_id: str | None = None
    name: str | None = None
    version_number: int | None = None
    effective_date: str | None = None
    settings: dict[str, Any] = Field(default_factory=dict)


class InternalDossierProposal(BaseModel):
    id: str
    locale: str
    status: str
    include_poc: bool
    assessment: dict[str, Any] | None = None
    proposal_body: dict[str, Any] | None = None
    poc: dict[str, Any] | None = None


class InternalDossierResponse(BaseModel):
    estimate_id: str
    project_name: str
    client_name: str
    status: str
    locale: str
    has_calculation: bool
    rate_card_stale: bool
    warnings: list[str] = Field(default_factory=list)
    report: dict[str, Any]
    rate_card: InternalDossierRateCard | None = None
    proposals: list[InternalDossierProposal] = Field(default_factory=list)
